app/models/concerns/hydranorth/generic_file/doi_states.rb

```diff
@@ -54,15 +54,21 @@ def withdraw_doi
         end
 
         def handle_doi_states
-          return unless doi_fields_present?
-          # TODO: handle unminted/unsynced states?
```

Comment:
New added logic below handles the case of unminted/unsynced states. So if a resque job ever fails and a generic file ends up in one of these states, it's possible to resave the generic file to create a new job to update/create a DOI. Instead of forever banishing them from getting a DOI.